setup.py

```diff
@@ -18,6 +22,73 @@
 
 name = environ.get("LIBRARY_NAME", "reco_utils")
 
+install_requires = [
+    "matplotlib>=2.2.2,<3",
+    "memory_profiler>=0.54.0,<1",
+    "nni==1.5",
+    "numba>=0.38.1,<1",
+    "numpy>=1.13.3,<2",
+    "pandas>1.0.3,<2",
+    "pydocumentdb>=2.3.3<3",  # todo: replace with azure-cosmos
+    "pyyaml>=5.4.1,<6",
+    "requests>=2.0.0,<3",
+    "seaborn>=0.8.1,<1",
+    "scipy>=1.0.0,<2",
+    "scikit-learn>=0.19.1,<1",
+    "tqdm>=4.31.1,<5",
+]
+
+extras_require = {
+    "recommenders": [
+        "bottleneck>=1.2.1,<2",
+        "category_encoders>=1.3.0,<2",
+        "cornac>=1.1.2,<2",
+        "jinja2>=2,<3",
+        "lightfm>=1.15,<2",
+        "lightgbm>=2.2.1,<3",
+        "nltk>=3.4,<4",
+        "pymanopt>=0.2.5,<1",
+        "scikit-surprise>=0.19.1,<2",
+        "transformers>=2.5.0,<5",
+    ],
+    "examples": [
+        "azure.mgmt.cosmosdb>=0.8.0,<1",
+        "hyperopt>=0.1.2,<1",
+        "ipykernel>=4.6.1,<5",
+        "jupyter>=1,<2",
+        "locust>=1,<2",
+        "papermill>=2.1.2,<3",
+        "scrapbook>=0.5.0,<1.0.0",        
+    ],
+    "gpu": [
+        "nvidia-ml-py3>=7.352.0",
+        "tensorflow-gpu==1.15.4",
+        "fastai>=1.0.46,<2",
+        "torch>=1.0.0,<2",
+    ],
+    "spark": [
+        "databricks_cli>=0.8.6,<1",
+        "pyarrow>=0.8.0,<1.0.0",
+        "pyspark>=2.4.5,<3.0.0",
+    ],
+    "test": [
+        "black>=18.6b4,<21",
+        "papermill>=2.1.2,<3",
+        "pytest>=3.6.4",
+        "scrapbook>=0.5.0,<1.0.0",
+    ],
+}
+# for the brave of heart
+extras_require["all"] = list(set(sum([*extras_require.values()], [])))
+
+# the following dependencies need additional testing
+extras_require["beta"] = [
+    "azureml-sdk[notebooks,tensorboard]>=1.0.69,<2",
```

Comment:
Starting a thread for discussing azureml, one question that was asked by @anargyri: What are the utilities and notebooks using azureml?

Reply:
I found the following (under examples/):

run_notebook_on_azureml.ipynb
00_quick_start/sar_movielens_with_azureml.ipynb
00_quick_start/sar_movieratings_with_azureml_designer.ipynb
04_model_select_and_optimize/azureml_hyperdrive_surprise_svd.ipynb
04_model_select_and_optimize/azureml_hyperdrive_wide_and_deep.ipynb
05_operationalize/aks_locust_load_test.ipynb
05_operationalize/als_movie_o16n.ipynb
05_operationalize/lightgbm_criteo_o16n.ipynb

Reply:
Also files in 
reco_utils/azureml/

Reply:
yep, outside or reco-utils/azureml I believe these are the only references

Reply:
if we look into `reco_utils/azureml`, the only utility is actually [this file](https://github.com/microsoft/recommenders/blob/main/reco_utils/azureml/azureml_utils.py), which is only the function `get_or_create_workspace`. 

The other notebooks or files are either independent of azureml or use azureml API, but are not utilities. 

So do we want to install azureml together with reco_utils or do we want to split these dependencies and when people execute one of the examples we ask the user to install azureml independently?

Reply:
right, wrapping workspace creation seems unnecessary, we can do that in the notebooks
also, the other functions seem to be relevant only for certain notebook examples, so I think we should move them out of the reco_utils folder and into the notebooks (or in a subfolder of the notebook where it is used?)
we can also move aks_utils -> reco_utils/common/k8s_utils.py
not sure what do do with the azureml_designer_modules package though? do we keep it here? move it to contrib?

Reply:
They relate to Hyperdrive, which requires the python script to be in a separate file, so it cannot be part of the Jupyter notebook itself. 
You could move the .py files under 04_model_select_and_optimize/ (even better inside a subfolder). I vaguely recall we chose the current configuration so that the notebook directories look more neat and are separate from the required utils. 
An alternative is to create an azureml folder under reco_utils/tuning, since NNI scripts are already there anyway.
Bear in mind that you will need to modify the notebooks to point to the new location of the python file.

Reply:
I guess what Miguel is hinting at is move azureml/ outside reco_utils?

Reply:
yeah

Reply:
makes sense, do you want to drop it from the experimental dependencies? I've moved out everything under reco_utils.azureml except for azureml_designer_modules package, what do you want to do with that?